includes/core/server/request/trait-use-test-mode-only-when-dev-mode.php

```diff
@@ -0,0 +1,24 @@
+<?php
+/**
+ * Trait file for WCPay\Core\Server\Request\Use_Test_Mode_Only_When_Dev_Mode.
+ *
+ * @package WooCommerce Payments
+ */
+
+namespace WCPay\Core\Server\Request;
+
+use WC_Payments;
+
+/**
+ * Trait for requests that we only send a test mode request if in dev mode.
+ */
+trait Use_Test_Mode_Only_When_Dev_Mode {
+	/**
+	 * Set the test_mode parameter to true only if in dev mode.
+	 *
+	 * @return void
+	 */
+	public function set_test_mode_only_when_dev_mode() {
```

Comment:
I would rename this function to simply `set_test_mode` or `add_test_mode_parameter` to avoid confusion, since the function name says something like that test mode will be set only when dev mode is enabled.

Reply:
I'm wondering whether this should be approached through an external method at all. The way the request worked within the API client did not allow the mode to be set externally, and maybe we should keep it similar. How about extending the `get_default_params` method within `Get_Account`, and including this straight in defaults?

Reply:
@RadoslavGeorgiev - yeah, I would agree the point of avoiding an external method at all, and proposed extending the constructor to set_param `test_mode` right there https://github.com/Automattic/woocommerce-payments/pull/5879#discussion_r1155710619, which would be even more simple and straightforward than extend `get_default_params`. 

However, its readability is reduced and the idea of overriding the constructor in a trait is not really nice. Therefore, I went with the current approach, though it requires an external method.

Reply:
> I would rename this function to simply set_test_mode or add_test_mode_parameter to avoid confusion, since the function name says something like that test mode will be set only when dev mode is enabled.

@zmaglica - I tried to balance this approach :D Your proposed names do not explicitly the relevance with `dev_mode` so I am using this name `set_test_mode_according_to_dev_mode` - commit c6e5724a3e58ba8009c5921426f90332eac2bed2

Reply:
> However, its readability is reduced and the idea of overriding the constructor in a trait is not really nice. Therefore, I went with the current approach, though it requires an external method.

I do agree that `get_default_params` should be obfuscated within a trait, however I'm not sure it's enough code to warrant a trait, especially if it could be automatically done within the request:

```php
public static function get_default_params() {
	$params              = parent::get_default_params();
	$params['test_mode'] = WC_Payments::mode()->is_dev(); // Explain why.
	return $params;
}
```

Using `get_default_params` to append a dynamic default parameter seems like very straight-forward and logical behavior anyway.

Reply:
> however I'm not sure it's enough code to warrant a trait, especially if it could be automatically done within the request. 

The code is pretty simple, but I'd prefer to use trait as currently, there are 4 methods using this only dev-mode pattern in class `WC_Payments_API_Client` - [1](https://github.com/Automattic/woocommerce-payments/blob/4c2f691c6e0fda63292fa20385a99350da4216f7/includes/wc-payment-api/class-wc-payments-api-client.php#L861-L861), [2](https://github.com/Automattic/woocommerce-payments/blob/4c2f691c6e0fda63292fa20385a99350da4216f7/includes/wc-payment-api/class-wc-payments-api-client.php#L878-L878), [3](https://github.com/Automattic/woocommerce-payments/blob/4c2f691c6e0fda63292fa20385a99350da4216f7/includes/wc-payment-api/class-wc-payments-api-client.php#L897-L897), [4](https://github.com/Automattic/woocommerce-payments/blob/4c2f691c6e0fda63292fa20385a99350da4216f7/includes/wc-payment-api/class-wc-payments-api-client.php#L1032-L1032). IMO, using trait here will increase readability and reusability

> Using `get_default_params` to append a dynamic default parameter seems like very straight-forward and logical behavior anyway.

After looking at this again, I agree this point and better than overriding the parent class `Request`'s constructor. WDYT about this approach @zmaglica?

Reply:
I like Rado's approach too, but since we have multiple request classes that use the same parameter, maybe we can use the best of both worlds. Inside Trait, add code that Rado shared and use that trait inside these endpoints to load dev mode param.

Reply:
Great that we have the alignment here. Addressed 9d13dcf979483c0c5ae6fe304b5c54974479e174